pkg/server/types.go

```diff
@@ -7,10 +7,12 @@ import (
 )
 
 type Config struct {
-	DisableAgent     bool
-	DisableServiceLB bool
-	ControlConfig    config.Control
-	Rootless         bool
-	SupervisorPort   int
-	StartupHooks     []func(context.Context, <-chan struct{}, string) error
+	DisableAgent      bool
+	DisableServiceLB  bool
+	ControlConfig     config.Control
+	Rootless          bool
+	SupervisorPort    int
+	StartupHooks      []func(context.Context, <-chan struct{}, string) error
+	LeaderControllers []func(ctx context.Context, sc *Context) error
```

Comment:
Can we use the CustomControllers type here so we don't have to manage declaration in 2 different places? If this creates a cycle, we may want to think about where the type itself is declared.

Reply:
Good point. Originally I did have an import cycle, but let me try moving the `CustomControllers` type to this package and import it back to the `cli/server` package.